File: examples/java-ddd/infra/src/main/java/org/myddd/java/domain/repository/UserRepositoryJPA.java
package org.myddd.java.domain.repository;

import org.myddd.java.domain.User;
import org.myddd.java.domain.UserRepository;
import org.myddd.persistence.jpa.AbstractRepositoryJPA;

import javax.inject.Named;
import java.util.List;

@Named
public class UserRepositoryJPA extends AbstractRepositoryJPA implements UserRepository {
    @Override
    public User createUser(User user) {
        return save(user);
    }

    @Override
    public List<User> searchUserByName(String name) {
        return getEntityManager()
                .createQuery("from User where name like :name",User.class)
                .setParameter("name","%" + name + "%")
                .getResultList();
    }
}

File: examples/kotlin-ddd/infra/src/main/kotlin/org/myddd/kotlin/infra/UserRepositoryJPA.kt
package org.myddd.kotlin.infra

import org.myddd.kotlin.domain.User
import org.myddd.kotlin.domain.UserRepository
import org.myddd.persistence.jpa.AbstractRepositoryJPA
import javax.inject.Named

@Named
class UserRepositoryJPA:AbstractRepositoryJPA(),UserRepository {

    override fun createUser(user: User): User?  = save(user)

    override fun searchUserByName(name: String): List<User> {
        return entityManager
            .createQuery("from User where name like :name",User::class.java)
            .setParameter("name","%$name%")
            .resultList
    }
}